Dockerfile

```diff
@@ -1,28 +1,74 @@
 # Docker Container that runs the Nebra Diagnostics Tool
 
-FROM balenalib/raspberry-pi-debian-python:buster-run-20211014
+####################################################################################################
+################################## Stage: builder ##################################################
 
+# The balenalib/raspberry-pi-debian-python image was tested but missed many dependencies.
+FROM balenalib/raspberry-pi-debian:buster-build-20211014 as builder
+
+# Nebra uses /opt by convention
 WORKDIR /opt/
 
+# Copy python dependencies for `pip install` later
+COPY requirements.txt requirements.txt
+
+# This will be the path that venv uses for installation below
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Install python3-minimal, pip3, wget, venv.
+# Then set venv environment copied from builder.
+# Finally, use pip to install dependencies.
+RUN \
+    install_packages \
+            python3-dev=3.7.3-1 \
+            python3-minimal=3.7.3-1 \
+            python3-pip=18.1-5+rpt1 \
+            python3-venv=3.7.3-1 \
+            libgirepository1.0-dev=1.58.3-2 \
+            gcc=4:8.3.0-1+rpi2 \
+            pkg-config=0.29-6 \
+            libdbus-1-3 && \
+    # Because the PATH is already updated above, this command creates a new venv AND activates it
+    python3 -m venv /opt/venv && \
```

Comment:
IIRC @marvinmarnold and @vpetersson mentioned previously that there was little point in using venv in containerised setup.

Think was in relation to hm-config but guess it applies here too

Reply:
I still have a soft spot for venv since it makes python portable. But yea, we did agree to stop using it.

@posterzh hm-pktfwd is an example of the new pattern: https://github.com/NebraLtd/hm-pktfwd/blob/e067bef1be17f9d3c1036cbe1100b317d5cc5f57/Dockerfile#L65

Reply:
Nice pattern.

Reply:
@marvinmarnold I am not fussed either way! Whichever you think is best TBH